service/db.go

```diff
@@ -88,13 +86,30 @@ func refreshMongoDBConnection(client *mongo.Client, uri *string) error {
 			}
 			return errMongoDBUnavailable
 		}
-		if strings.EqualFold(*uri, conf.DocumentDB.Reader) {
-			mongoDBReader = newClient
-		} else {
-			mongoDBWriter = newClient
+		if err := assignMongoDBClient(newClient, uri); err != nil {
+			return err
 		}
 		return nil
 	}
 
 	return nil
 }
+
+// assignMongoDBClient assigns a new MongoDB client.
+// Returns an error if newClient is nil.
+func assignMongoDBClient(newClient *mongo.Client, uri *string) error {
```

Comment:
I guess I was picturing this differently...
I was picturing it done this way: 

```
func assignMongoDBClient(newClient *mongo.Client, uri *string) {
        if strings.EqualFold(*uri, conf.DocumentDB.Reader) {
 		mongoDBReader = newClient
 	} else {
 		mongoDBWriter = newClient
 	}
 }
```

And it will be used two ways:

```
assignMongoDBClient(newClient, uri)
```

or 

```
assignMongoDBClient(nil, uri)
```

The purpose of this was to remove the redundancy, but it is still there ( i can see why you didn't use it b/c it will be triple nested) so maybe this wasn't a good idea from the first place... I'll leave the option up to you.

Reply:
I am a pessimistic programmer.

Reply:
I removed the redundancy. Thank you for reviewing.